core/storage/audit/gae_models.py

```diff
@@ -0,0 +1,42 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing the audit logs."""
+
+import feconf
+from core.platform import models
+from google.appengine.ext import ndb
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
+
+
+class RoleQueryAuditModel(base_models.BaseModel):
+    """Records the data for query made to the role structure using admin
+    interface.
+    """
+    # The user_id of the user making query.
+    user_id = ndb.StringProperty(required=True, indexed=True)
+    # The intent of making query (viewing role or updating role).
+    intent = ndb.StringProperty(required=True, choices=[
```

Comment:
Why ACTION in name? ROLE_UPDATE, ROLE_VIEW_BY_USERNAME, ROLE_VIEW_BY_ROLE sounds fine for intent?